File: turnos-app-frontend/src/types/Permiso.ts
// Coincide con el enum Permiso del backend (src/Core.Domain/Enums/Permiso.cs).
// Agrupado por área para la grilla de checkboxes de RolesPage.
export type Permiso =
  | 'VerAgendaCompleta'
  | 'GestionarTurnos'
  | 'GestionarClientes'
  | 'GestionarServicios'
  | 'GestionarRecursos'
  | 'GestionarMetodosPago'
  | 'CrearCobros'
  | 'VerGananciaNeta'
  | 'VerReportes'
  | 'GestionarUsuarios'
  | 'GestionarRoles'
  | 'GestionarConfiguracionNegocio'
  | 'GestionarListaEspera'
  | 'GestionarCaja'
  | 'ForzarCierreCaja'
  | 'VerCaja'
  | 'VerLiquidaciones'
  | 'GestionarLiquidaciones'
  | 'GestionarSuscripcion';

export const PERMISOS_POR_AREA: { area: string; permisos: { valor: Permiso; label: string }[] }[] = [
  {
    area: 'Turnos y agenda',
    permisos: [
      { valor: 'VerAgendaCompleta', label: 'Ver la agenda de todos los recursos (sin esto, solo ve la propia)' },
      { valor: 'GestionarTurnos', label: 'Crear, editar y cancelar turnos' },
      { valor: 'GestionarListaEspera', label: 'Anotar y gestionar la lista de espera' },
    ],
  },
  {
    area: 'Clientes y catálogo',
    permisos: [
      { valor: 'GestionarClientes', label: 'Gestionar clientes' },
      { valor: 'GestionarServicios', label: 'Gestionar servicios' },
      { valor: 'GestionarRecursos', label: 'Gestionar recursos (profesionales, salas)' },
    ],
  },
  {
    area: 'Cobros y finanzas',
    permisos: [
      { valor: 'CrearCobros', label: 'Registrar y editar cobros' },
      { valor: 'GestionarMetodosPago', label: 'Gestionar métodos de pago' },
      { valor: 'VerGananciaNeta', label: 'Ver la ganancia neta del negocio' },
      { valor: 'VerReportes', label: 'Ver reportes y métricas' },
    ],
  },
  {
    area: 'Equipo y negocio',
    permisos: [
      { valor: 'GestionarUsuarios', label: 'Gestionar usuarios del equipo' },
      { valor: 'GestionarRoles', label: 'Gestionar roles y permisos' },
      { valor: 'GestionarConfiguracionNegocio', label: 'Gestionar la configuración del negocio' },
    ],
  },
  {
    area: 'Caja',
    permisos: [
      { valor: 'VerCaja', label: 'Ver el estado de caja y el historial de sesiones (sin poder operarla)' },
      { valor: 'GestionarCaja', label: 'Abrir, cerrar y registrar movimientos de caja (incluye poder verla)' },
      { valor: 'ForzarCierreCaja', label: 'Forzar el cierre de una sesión de caja abierta por otro usuario' },
    ],
  },
  {
    area: 'Liquidaciones',
    permisos: [
      { valor: 'VerLiquidaciones', label: 'Ver las liquidaciones de todos los profesionales (sin poder operarlas)' },
      { valor: 'GestionarLiquidaciones', label: 'Configurar comisiones, marcar liquidaciones pagadas, anularlas y cargar adelantos' },
    ],
  },
  {
    area: 'Suscripción',
    permisos: [
      { valor: 'GestionarSuscripcion', label: 'Gestionar la suscripción del negocio (checkout, portal de facturación)' },
    ],
  },
];

File: src/Core.Domain/Enums/Permiso.cs
using System;

namespace TurnosApp.Core.Domain.Enums;

[Flags]
public enum Permiso : long
{
    Ninguno = 0,

    // Sin este permiso, el usuario queda acotado a sus propios turnos (los de su Recurso vinculado).
    VerAgendaCompleta = 1L << 0,
    GestionarTurnos = 1L << 1,
    GestionarClientes = 1L << 2,
    GestionarServicios = 1L << 3,
    GestionarRecursos = 1L << 4,
    GestionarMetodosPago = 1L << 5,
    CrearCobros = 1L << 6,
    VerGananciaNeta = 1L << 7,
    VerReportes = 1L << 8,
    GestionarUsuarios = 1L << 9,
    GestionarRoles = 1L << 10,
    GestionarConfiguracionNegocio = 1L << 11,
    GestionarListaEspera = 1L << 12,
    GestionarCaja = 1L << 13,
    ForzarCierreCaja = 1L << 14,
    VerCaja = 1L << 15,

    // Sin este permiso (ni GestionarLiquidaciones), un Recurso vinculado a Usuario igual
    // puede ver sus propias liquidaciones — es el mismo criterio de "mis turnos" self-service.
    VerLiquidaciones = 1L << 16,
    GestionarLiquidaciones = 1L << 17,

    // Crear/gestionar la suscripción del tenant (iniciar/cancelar el preapproval de Mercado Pago).
    GestionarSuscripcion = 1L << 18,

    Todos = VerAgendaCompleta | GestionarTurnos | GestionarClientes | GestionarServicios
        | GestionarRecursos | GestionarMetodosPago | CrearCobros | VerGananciaNeta
        | VerReportes | GestionarUsuarios | GestionarRoles | GestionarConfiguracionNegocio
        | GestionarListaEspera | GestionarCaja | ForzarCierreCaja | VerCaja
        | VerLiquidaciones | GestionarLiquidaciones | GestionarSuscripcion
}
